Realm/Realm.SourceGenerator/Analytics.cs

```diff
@@ -0,0 +1,524 @@
+            { SdkFeature.RecoverOrDiscardUnsyncedChangesHandler, 0 },
+            { SdkFeature.RecoverUnsyncedChangesHandler, 0 },
+            { SdkFeature.DiscardUnsyncedChangesHandler, 0 },
+            { SdkFeature.ManualRecoveryHandler, 0 },
+            { SdkFeature.GetProgressObservable, 0 },
+            { SdkFeature.PartitionSyncConfiguration, 0 },
+            { SdkFeature.FlexibleSyncConfiguration, 0 },
+            { SdkFeature.Anonymous, 0 },
+            { SdkFeature.EmailPassword, 0 },
+            { SdkFeature.Facebook, 0 },
+            { SdkFeature.Google, 0 },
+            { SdkFeature.Apple, 0 },
+            { SdkFeature.JWT, 0 },
+            { SdkFeature.ApiKey, 0 },
+            { SdkFeature.ServerApiKey, 0 },
+            { SdkFeature.Function, 0 },
+            { SdkFeature.CallAsync, 0 },
+            { SdkFeature.FIX_ME_THREE, 0 },
+            { SdkFeature.DynamicApi, 0 },
+        };
+
+        private Dictionary<string, string> _realmEnvMetrics = new()
+        {
+            { SdkFeature.UserId, string.Empty },
+            { SdkFeature.RealmSdk, ".NET" },
+            { SdkFeature.Language, "C#" },
+            { SdkFeature.HostOsType, string.Empty },
+            { SdkFeature.HostOsVersion, string.Empty },
+        };
+
+        private static string AnonymizedUserID
+        {
+            get
+            {
+                try
+                {
+                    var id = GenerateComputerIdentifier();
+                    return id != null ? SHA256Hash(id) : "UNKNOWN";
+                }
+                catch
+                {
+                    return "UNKNOWN";
+                }
+            }
+        }
+
+        public void SubmitAnalytics(GeneratorExecutionContext context)
+        {
+            var payload = string.Empty;
+
+            if (Environment.GetEnvironmentVariable("REALM_DISABLE_ANALYTICS") != null ||
+                Environment.GetEnvironmentVariable("CI") != null)
+            {
+                return;
+            }
+
+            _realmEnvMetrics[SdkFeature.UserId] = AnonymizedUserID;
+
+            ComputeHostOSNameAndVersion(out var osName, out var osVersion);
+
+            // TODO andrea: check if searching strings in this way works on non Windows platforms
+            switch (osName)
+            {
+                case string name when name.Contains(Metric.OperatingSystem.Windows):
+                    _realmEnvMetrics[SdkFeature.HostOsType] = Metric.OperatingSystem.Windows;
+                    break;
+                case string name when name.Contains(Metric.OperatingSystem.MacOS):
+                    _realmEnvMetrics[SdkFeature.HostOsType] = Metric.OperatingSystem.MacOS;
+                    break;
+                case string name when name.Contains(Metric.OperatingSystem.Linux):
+                    _realmEnvMetrics[SdkFeature.HostOsType] = Metric.OperatingSystem.Linux;
+                    break;
+                default:
+                    Diagnostics.Add(
```

Comment:
Same comment for the other diagnostics

Reply:
This is about SGs. We've now moved this to the weaver.